flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mongodb-cdc/src/main/java/org/apache/flink/cdc/connectors/mongodb/source/reader/fetch/MongoDBStreamFetchTask.java

```diff
@@ -283,6 +301,9 @@ private MongoChangeStreamCursor<BsonDocument> openChangeStreamCursor(
                         e.getErrorMessage(),
                         e.getErrorCode());
                 throw new FlinkRuntimeException("Unauthorized $changeStream operation", e);
+            } else if (timestamp != null && MongoUtils.checkIfResumeTokenExpires(e)) {
+                ((ChangeStreamOffset) streamSplit.getStartingOffset()).clearResumeToken();
+                return openChangeStreamCursor(changeStreamDescriptor);
```

Comment:
Recursively calling `openChangeStreamCursor` is safe here, since resume token is cleared here, and `checkResumeTokenExpires` would never be true again since no resume token is provided.

Reply:
I prefer to add a parameter to `openChangeStreamCursor` to indicate forcing the use of timestamp mode instead of clearing the resumeToken.
WDYT?

Reply:
Yes, that would be cleaner to add a switch parameter instead of keeping global variable / status elsewhere. Pushed another commit to solve that.